File: python/get_crypto_book_asks.py
from __future__ import print_function
import time
import intrinio_sdk
from intrinio_sdk.rest import ApiException
from pprint import pprint
 
intrinio_sdk.ApiClient().configuration.api_key['api_key'] = 'YOUR_API_KEY'
 
crypto_api = intrinio_sdk.CryptoApi()
 
pair = 'btcusd' # str | Return the order book asks for the given Crypto Currency Pair. (optional)
exchange = 'gemini' # str | Return the order book asks for a Crypto Currency on the given Crypto Exchange. (optional)
currency = 'BTC' # str | Return the order book asks for the given Crypto Currency. (optional)
 
try:
    api_response = crypto_api.get_crypto_book_asks(pair=pair, exchange=exchange, currency=currency)
    asks = api_response.asks

    print("Crypto Currency Pair: %s" % (api_response.pair.name))
    print("Crypto Exchange: %s" % (api_response.exchange.name))
    print("")
    print("-------------------------------------- ASKS -------------------------------------- ")

    for ask in asks:
        print("")
        print("Price: %s" % (ask.price))
        print("Size:  %s" % (ask.size))

except ApiException as e:
    print("Exception when calling CryptoApi->get_crypto_book_asks: %s\n" % e)

File: csharp/GetCryptoBookAsks.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using Intrinio.SDK.Api;
using Intrinio.SDK.Client;
using Intrinio.SDK.Model;

namespace Example
{
  public class GetCryptoBookAsks
  {
    public static void Main()
    {
      Configuration.Default.AddApiKey("api_key", "YOUR_API_KEY");

      var cryptoApi = new CryptoApi();
      var pair = "btcusd";  // string | Return the order book asks for the given Crypto Currency Pair. (optional) 
      var exchange = "gemini";  // string | Return the order book asks for a Crypto Currency on the given Crypto Exchange. (optional) 
      var currency = "BTC";  // string | Return the order book asks for the given Crypto Currency. (optional) 

      try
      {
        ApiResponseCryptoBookAsks result = cryptoApi.GetCryptoBookAsks(pair, exchange, currency);
        List<CryptoAsk> crypto_asks = result.Asks;

        CryptoPairSummary crypto_pair = result.Pair;
        Console.WriteLine("Crypto Currency Pair: " + crypto_pair.Name);

        CryptoExchangeSummary crypto_exchange = result.Exchange;
        Console.WriteLine("Crypto Exchange: " + crypto_exchange.Name);

        Console.WriteLine();
        Console.WriteLine("----------------- ASKS -----------------");

        crypto_asks.ForEach(delegate (CryptoAsk ask)
        {
          Console.WriteLine();
          Console.WriteLine("Price: " + ask.Price);
          Console.WriteLine("Size:  " + ask.Size);
        });
      }
      catch (Exception e)
      {
        Debug.Print("Exception when calling CryptoApi.GetCryptoBookAsks: " + e.Message);
      }
    }
  }
}
